hw/mcu/dialog/da1469x/src/hal_i2c.c

```diff
@@ -442,12 +442,13 @@ hal_i2c_master_read(uint8_t i2c_num, struct hal_i2c_master_data *pdata,
 int
 hal_i2c_master_probe(uint8_t i2c_num, uint8_t address, uint32_t timo)
 {
-    struct hal_i2c_master_data rx;
-    uint8_t buf;
+    struct hal_i2c_master_data tx;
+    uint8_t buf = 0;
 
-    rx.address = address;
-    rx.buffer = &buf;
-    rx.len = 1;
+    tx.address = address;
+    tx.buffer = &buf;
+    tx.len = 1;
 
-    return hal_i2c_master_read(i2c_num, &rx, timo, 1);
+    /* Reading before writing causes i2c bus to hang up sometimes. */
```

Comment:
I think this comment should explain more clearly why hal_i2c_master_write() is used instead of a read. Something like,  'Using a write because a read results in ... <whatever the issue with read was>'